cmd/aad-cli/cli/user_test.go

```diff
@@ -0,0 +1,275 @@
+package cli_test
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/aad-auth/cmd/aad-cli/cli"
+	"github.com/ubuntu/aad-auth/internal/cache"
+	"github.com/ubuntu/aad-auth/internal/testutils"
+	"golang.org/x/exp/slices"
+)
+
+func TestUserShellCompletion(t *testing.T) {
+	tests := map[string]struct {
+		args string
+	}{
+		"get all users, short flag":            {args: "user -n"},
+		"get all users, long flag":             {args: "user --name"},
+		"get attributes for user":              {args: "user"},
+		"get attributes for overridden user":   {args: "user --name futureuser@domain.com"},
+		"default completion for last argument": {args: "user gecos"},
+		"default completion, overridden user":  {args: "user gecos --name futureuser@domain.com"},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			args := []string{cobra.ShellCompRequestCmd}
+			args = append(args, strings.Split(tc.args, " ")...)
+			args = append(args, "")
+
+			cacheDir := t.TempDir()
+			cacheDB := "users_in_db"
+			testutils.PrepareDBsForTests(t, cacheDir, cacheDB)
+			cache := testutils.NewCacheForTests(t, cacheDir)
+
+			c := cli.New(cli.WithCache(cache))
+			got, err := testutils.RunApp(t, c, args...)
+			require.NoError(t, err, "failed to run completion")
+
+			want := testutils.LoadWithUpdateFromGolden(t, got)
+			require.Equal(t, want, got, "expected output to match golden file")
+		})
+	}
+}
+
+func TestUser(t *testing.T) {
+	tests := map[string]struct {
+		args               string
+		shadowNotAvailable bool
+
+		wantErr bool
+	}{
+		"get all users":                  {args: "--all"},
```

Comment:
So, the only can we don’t test this way is for the current user detection. Any idea how we could do this (as it’s not in db)?

Reply:
I meant "on the tests", like the detection of the current user. But it seems overkill TBH, wdyt?